server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -744,6 +744,11 @@ public int compare(File o1, File o2)
               )
           );
         }
+        if (hydrants.isEmpty()) {
+          // Probably encountered a corrupt sink directory
+          log.warn("Found persisted sink with no hydrants at %s, skipping sink creation.", sinkDir.getAbsolutePath());
```

Comment:
Can we instead say. Found persisted segment directory with no intermediate segments present